imap_processing/ultra/l1b/de.py

```diff
@@ -321,7 +322,7 @@ def calculate_de(
     # TODO - find a better solution than filtering out data from repointings?
     if repoint_id is not None:
         in_pointing = calculate_events_in_pointing(
-            repoint_id, event_times, valid_events
+            repoint_id, event_times, quality_flags
```

Comment:
Rather than calculate_events_in_pointing doing the quality_flags update internally, do you want to handle that outside here. I think then the function would be more self-contained and potentially usable in other instruments (future) and mean you don't need to add a zeros pass-through in the test.

Then below this, why do you have a "valid_events" array, isn't that the same as what quality_flags is meant to indicate? i.e. I think we should be updating the quality flags throughout all these routines and then at the end use that to figure out what events are valid or not based upon the quality flags.